docs/technical-doc/Endorsements.rst

```diff
@@ -0,0 +1,118 @@
+============
+Endorsements
+============
+
+Intro
+=====
+
+
+Massa uses the Proof-of-Stake selection mechanism with Nakamoto consensus.
+In that context, when there are multiple cliques in close competition,
+we want all nodes to converge towards a single clique as fast as possible to minimize finality time and maximize the quality of the consensus.
+To achieve this, we draw inspiration from Tezos and introduce the concept of Endorsement.
+
+Basic principle
+===============
+
+At every slot `S`, we use the existing Proof-of-Stake selection mechanism to not only draw the block creator for that slot,
+but also `E` other stakers that are responsible for creating Endorsements that can be included in the blocks of slot `S`.
+Every endorsement that is selected to be included in blocks of slot `S` is signed by its creator `C`
+and contains the "endorsed" hash of the block that `C` would choose as the parent of a block at slot `S` in thread `S.thread`.
+Blocks in slot `S` can only include endorsements that endorse the block's parent block in the thread of `S`.
+In other words, endorsement producers vote for which parent should be chosen by blocks in their own thread.
+
+The likelihood of the attacker getting lucky and being selected for `N` consecutive PoS draws to attack/censor the system decays exponentially with `N`.
+With endorsements, we don't have to wait for `N` blocks to account for `N` proof-of-stake draws to happen as `E+1` draws happen at every slot.
+In the consensus algorithm, we choose the clique of highest fitness as the blockclique.
+A block including `e` endorsements out of the maximum `E` contributes a fitness `e + 1` to the cliques it belongs to.
+The fitness of a block is therefore reflected by the number of PoS draws that were involved in creating it.
+
+The net effect of this mechanism is to increase safety and convergence speed
+by allowing block producers to quickly choose the best clique to extend according to the "votes" provided by the endorsements.
+
+
+Structure of an endorsement
+===========================
+
+.. code:: rust
+
+  pub struct Endorsement {
+    /// slot in which the endorsement can be included
+    pub slot: Slot,
+    /// endorsement index inside the including block
+    pub index: u32,
+    /// hash of endorsed block
+    pub endorsed_block: BlockId,
```

Comment:
We can precise that the endorsed block is the selected parent in the thread of the slot. Selected by the producer of the endorsement.

Reply:
added